test/types.c

```diff
@@ -960,6 +960,54 @@ ODBC_TEST(t_bindsqlnum_basic)
 }
 
 
+/*
+   Test of binding a SQL_NUMERIC_STRUCT as a query parameter
+   with more precision than is really needed
+*/
+ODBC_TEST(t_bindsqlnum_wide)
+{
+  SQL_NUMERIC_STRUCT *sqlnum= malloc(sizeof(SQL_NUMERIC_STRUCT));
+  SQLCHAR outstr[20];
+  memset(sqlnum, 0, sizeof(SQL_NUMERIC_STRUCT));
+
+  sqlnum->sign= 1;
+  sqlnum->scale= 3;
+  sqlnum->val[0]= 0x7c;
+  sqlnum->val[1]= 0x62;
+
+  CHECK_HANDLE_RC(SQL_HANDLE_STMT, Stmt, SQLPrepare(Stmt, (SQLCHAR *)"select ?", SQL_NTS));
+
+  CHECK_HANDLE_RC(SQL_HANDLE_STMT, Stmt, SQLBindParameter(Stmt, 1, SQL_PARAM_INPUT, SQL_C_NUMERIC,
+                                  SQL_DECIMAL, 15, 3,
+                                  sqlnum, 0, NULL));
+
+  CHECK_HANDLE_RC(SQL_HANDLE_STMT, Stmt, SQLExecute(Stmt));
+  CHECK_HANDLE_RC(SQL_HANDLE_STMT, Stmt, SQLFetch(Stmt));
+  CHECK_HANDLE_RC(SQL_HANDLE_STMT, Stmt, SQLGetData(Stmt, 1, SQL_C_CHAR, outstr, 20, NULL));
+  IS_STR(outstr, "25.212", 6);
+  is_num(sqlnum->sign, 1);
+  is_num(sqlnum->precision, 15);
```

Comment:
This is disturbing me. You check here program's bound numeric structure after execution. I can see that the driver puts precision and scale values there, but should it really? I think the general rule is not to touch application's memory (unless that is designated for that buffers). I glanced over ODBC docs, and couldn't find justification of that. Do you happen to know that justification and couldn't you point to ODBC specs? Otherwise I think this part has to be removed from the test, and I should fix the driver accordingly

Reply:
I agree that is odd - it was basically copied from the existing `t_bindsqlnum_basic` test and then modified to increase the precision of the bind and to include a negative value.